deepspeed/pt/deepspeed_cuda.py

```diff
@@ -393,21 +401,24 @@ def backward(ctx, grad_output):
 class DeepSpeedTransformerLayer(nn.Module):
     """Initialize the DeepSpeed Transformer Layer.
 
+        Static variable:
+            layer_id: The layer-index counter starting from 0 and incrementing by 1 every time a layer object is instantiated,
+            e.g. if a model has 24 transformer layers, layer_id goes from 0 to 23.
         Arguments:
-            layer_id: The layer index starting from 0, e.g. if model has 24 transformer layers,
-                layer_id will be 0,1,2...23 when each layer object is instantiated
-
             config: An object of DeepSpeedTransformerConfig
 
             initial_weights: Optional: Only used for unit test
 
             initial_biases: Optional: Only used for unit test
     """
-    def __init__(self, layer_id, config, initial_weights=None, initial_biases=None):
+    layer_id = 0
+
+    def __init__(self, config, initial_weights=None, initial_biases=None):
```

Comment:
Can the unit tests instead allocate the layer and then replace the weights/biases? I would prefer to keep these details from complicating our API. They also result in a big branch in the code just below.

Reply:
The issue here is that we need to compare our implementation with baseline, and if we don't get it from upper level, there is no way that we have the same weights/biases. I remember we tried it in the beginning with @eltonzheng and we could not come up with any other solution of having similar weight as PyTorch. We even tried the same random seed but still the weights between our case and PyTorch was different!